pkg/resourcemanager/psql/psqluser/psqluser.go

```diff
@@ -0,0 +1,285 @@
+		ctx,
+		secretKey,
+	)
+	if err != nil {
+		instance.Status.Message = "failed to delete secret, err: " + err.Error()
+		return false, err
+	}
+
+	// delete all the custom formatted secrets if keyvault is in use
+	keyVaultEnabled := reflect.TypeOf(secretClient).Elem().Name() == "KeyvaultSecretClient"
+	if keyVaultEnabled {
+		customFormatNames := []string{
+			"adonet",
+			"adonet-urlonly",
+			"jdbc",
+			"jdbc-urlonly",
+			"odbc",
+			"odbc-urlonly",
+			"server",
+			"database",
+			"username",
+			"password",
+		}
+
+		for _, formatName := range customFormatNames {
+			key := types.NamespacedName{Namespace: secretKey.Namespace, Name: instance.Name + "-" + formatName}
+
+			err = secretClient.Delete(
+				ctx,
+				key,
+			)
+			if err != nil {
+				instance.Status.Message = "failed to delete secret, err: " + err.Error()
+				return false, err
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// GetOrPrepareSecret gets or creates a secret
+func (s *PostgreSqlUserManager) GetOrPrepareSecret(ctx context.Context, instance *v1alpha1.PostgreSQLUser, secretClient secrets.SecretClient) map[string][]byte {
+	key := GetNamespacedName(instance, secretClient)
+
+	secret, err := secretClient.Get(ctx, key)
+	if err != nil {
+		// @todo: find out whether this is an error due to non existing key or failed conn
+		pw := helpers.NewPassword()
+		return map[string][]byte{
+			"username":                 []byte(""),
+			"password":                 []byte(pw),
+			"PSqlServerNamespace":      []byte(instance.Namespace),
+			"PSqlServerName":           []byte(instance.Spec.Server),
+			"fullyQualifiedServerName": []byte(instance.Spec.Server + "." + config.Environment().SQLDatabaseDNSSuffix),
+			"PSqlDatabaseName":         []byte(instance.Spec.DbName),
+		}
+	}
+	return secret
+}
+
+// GetNamespacedName gets the namespaced-name
+func GetNamespacedName(instance *v1alpha1.PostgreSQLUser, secretClient secrets.SecretClient) types.NamespacedName {
+	var namespacedName types.NamespacedName
+	keyVaultEnabled := reflect.TypeOf(secretClient).Elem().Name() == "KeyvaultSecretClient"
+
+	if keyVaultEnabled {
+		// For a keyvault secret store, check for supplied namespace parameters
+		var dbUserCustomNamespace string
+		if instance.Spec.KeyVaultSecretPrefix != "" {
+			dbUserCustomNamespace = instance.Spec.KeyVaultSecretPrefix
+		} else {
+			dbUserCustomNamespace = "psqluser-" + instance.Spec.Server + "-" + instance.Spec.DbName
+		}
+
+		namespacedName = types.NamespacedName{Namespace: dbUserCustomNamespace, Name: instance.Name}
+	} else {
+		namespacedName = types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}
+	}
+
+	return namespacedName
+}
+
+func findBadChars(stack string) error {
+	badChars := []string{
+		"'",
+		"\"",
+		";",
+		"--",
+		"/*",
+	}
+
+	for _, s := range badChars {
+		if idx := strings.Index(stack, s); idx > -1 {
+			return fmt.Errorf("potentially dangerous character seqience found: '%s' at pos: %d", s, idx)
+		}
+	}
+	return nil
+}
```

Comment:
this function already exists so it should be moved to a shared location like in the helpers package